### ISSUE TITLE
The layout engine does not take border size into account when sizing_mode="stretch_width"

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#903]
#### System Info

- Panel: 0.7.0
- Bokeh: 1.4.0
- Tornado 5.1.11
- Python: 3.7.4
- OS: Windows 8.1
- Browser: Chrome

#### My Pain

I'm trying to develop a Card layout for my Bootstrap Template for awesome-panel.org. But it seems the the layout engine has problems when I set want the Card to have a border and the child items to be full width, responsive.

![image](https://user-images.githubusercontent.com/42288570/71348617-363f4300-256d-11ea-9f1c-c3629f0ed57a.png)

#### Reproducible Example

```python
import hvplot.pandas
import pandas as pd
import panel as pn


def _get_chart_data() -> pd.DataFrame:
    """## Chart Data

    Returns:
        pd.DataFrame -- A DataFrame with dummy data and columns=["Day", "Orders"]
    """

    chart_data = {
        "Day": ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday",],
        "Orders": [15539, 21345, 18483, 24003, 23489, 24092, 12034],
    }
    return pd.DataFrame(chart_data)


def _holoviews_chart():
    """## Dashboard Orders Chart generated by HoloViews"""
    data = _get_chart_data()
    line_plot = data.hvplot.line(
        x="Day", y="Orders", width=None, height=500, line_color="#007BFF", line_width=6,
    )
    scatter_plot = data.hvplot.scatter(x="Day", y="Orders", height=300).opts(
        marker="o", size=10, color="#007BFF"
    )
    fig = line_plot * scatter_plot
    gridstyle = {"grid_line_color": "black", "grid_line_width": 0.1}
    fig = fig.opts(
        responsive=True,
        toolbar=None,
        yticks=list(range(12000, 26000, 2000)),
        ylim=(12000, 26000),
        gridstyle=gridstyle,
        show_grid=True,
    )
    return fig


css = """
.bk.card {
    border: 10px solid rgba(0,0,0,.125);
    border-radius: 0.25rem;
}
.bk.card-header {
    background-color: rgba(0, 0, 0, 0.03);
    border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}
.bk.card-panel {
    background: salmon;
}
"""
pn.config.raw_css.append(css)
chart = _holoviews_chart()
chart_panel = pn.panel(chart, css_classes=["card-panel"], background="salmon", margin=0)
header = pn.pane.HTML(
    "<h5>header</h5>", css_classes=["card-header"], sizing_mode="stretch_width", margin=0
)
body = pn.pane.Markdown("this is my text. " * 50, css_classes=["card-panel"], margin=0)
card = pn.Column(
    header,
    body,
    chart_panel,
    sizing_mode="stretch_width",
    css_classes=["card"],
    style={"border": "10px solid rgba(0,0,0,.125)", "border-radius": "0.25rem"},
)

app = pn.Column(card, width=600)

app.servable("no css_classes")
```

**Comment by MarcSkovMadsen** [comment]
A **workaround** or **hack** is to insert the children inside a column with relevant margins.

But as the layout engine does not perform well if you have a lot of nested rows and columns this is not a good solution (I think).

![image](https://user-images.githubusercontent.com/42288570/71348790-9e8e2480-256d-11ea-8b00-889d7b3cc3e6.png)

#### Code

```python
import hvplot.pandas
import pandas as pd
import panel as pn


def _get_chart_data() -> pd.DataFrame:
    """## Chart Data

    Returns:
        pd.DataFrame -- A DataFrame with dummy data and columns=["Day", "Orders"]
    """

    chart_data = {
        "Day": ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday",],
        "Orders": [15539, 21345, 18483, 24003, 23489, 24092, 12034],
    }
    return pd.DataFrame(chart_data)


def _holoviews_chart():
    """## Dashboard Orders Chart generated by HoloViews"""
    data = _get_chart_data()
    line_plot = data.hvplot.line(
        x="Day", y="Orders", width=None, height=500, line_color="#007BFF", line_width=6,
    )
    scatter_plot = data.hvplot.scatter(x="Day", y="Orders", height=300).opts(
        marker="o", size=10, color="#007BFF"
    )
    fig = line_plot * scatter_plot
    gridstyle = {"grid_line_color": "black", "grid_line_width": 0.1}
    fig = fig.opts(
        responsive=True,
        toolbar=None,
        yticks=list(range(12000, 26000, 2000)),
        ylim=(12000, 26000),
        gridstyle=gridstyle,
        show_grid=True,
    )
    return fig


css = """
.bk.card {
    border: 10px solid rgba(0,0,0,.125);
    border-radius: 0.25rem;
}
.bk.card-header {
    background-color: rgba(0, 0, 0, 0.03);
    border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}
.bk.card-panel {
    background: salmon;
}
"""
pn.config.raw_css.append(css)
chart = _holoviews_chart()
chart_panel = pn.panel(chart, css_classes=["card-panel"], background="salmon", margin=0)
header = pn.pane.HTML(
    "<h5>header</h5>", css_classes=["card-header"], sizing_mode="stretch_width", margin=0
)
body = pn.pane.Markdown("this is my text. " * 50, css_classes=["card-panel"], margin=0)
card = pn.Column(
    pn.Column(header, body, chart_panel, sizing_mode="stretch_width", margin=(0, 20, 20, 0)),
    sizing_mode="stretch_width",
    css_classes=["card"],
    style={"border": "10px solid rgba(0,0,0,.125)", "border-radius": "0.25rem"},
)

app = pn.Column(card, width=600)

app.servable("no css_classes")
```

**Comment by MarcSkovMadsen** [comment]
Is this something that should be posted to Bokeh?

**Comment by philippjfr** [comment]
Closing since this is indeed a bokeh issue and should be resolved by bokeh 3.0
